scipy/differentiate/_differentiate.py

```diff
@@ -49,33 +53,47 @@ def _differentiate_iv(func, x, args, atol, rtol, maxiter, order, initial_step,
             step_factor, step_direction, preserve_shape, callback)
 
 
-def _differentiate(func, x, *, args=(), atol=None, rtol=None, maxiter=10,
-                   order=8, initial_step=0.5, step_factor=2.0,
-                   step_direction=0, preserve_shape=False, callback=None):
-    """Evaluate the derivative of an elementwise scalar function numerically.
+def differentiate(f, x, *, args=(), tolerances=None, maxiter=10,
+                  order=8, initial_step=0.5, step_factor=2.0,
+                  step_direction=0, preserve_shape=False, callback=None):
+    """Evaluate the derivative of a elementwise, real scalar function numerically.
+
+    For each element of the output of `f`, `differentiate` approximates the first
+    derivative of `f` at the corresponding element of `x` using finite difference
+    differentiation.
+
+    This function works elementwise when `x`, `step_direction`, and `args` contain
+    (broadcastable) arrays.
 
     Parameters
     ----------
-    func : callable
+    f : callable
         The function whose derivative is desired. The signature must be::
 
-            func(x: ndarray, *fargs) -> ndarray
+            f(x: ndarray, *fargs) -> ndarray
 
          where each element of ``x`` is a finite real number and ``fargs`` is a tuple,
          which may contain an arbitrary number of arrays that are broadcastable
-         with `x`. ``func`` must be an elementwise function: each element
-         ``func(x)[i]`` must equal ``func(x[i])`` for all indices ``i``.
-    x : array_like
-        Abscissae at which to evaluate the derivative.
-    args : tuple, optional
-        Additional positional arguments to be passed to `func`. Must be arrays
-        broadcastable with `x`. If the callable to be differentiated requires
-        arguments that are not broadcastable with `x`, wrap that callable with
-        `func`. See Examples.
-    atol, rtol : float, optional
-        Absolute and relative tolerances for the stopping condition: iteration
-        will stop when ``res.error < atol + rtol * abs(res.df)``. The default
-        `atol` is the smallest normal number of the appropriate dtype, and
+         with `x`. ``f`` must be an elementwise function: each element
```

Comment:
```suggestion
         with `x`. `f` must be an elementwise function: each element
```

Reply:
Haha yes but also
```suggestion
         with ``x``. `f` must be an elementwise function: each element
```
Because here `x` refers to the argument passed to `f`, which is not the same as the argument passed to `differentiate`. I'll give it a different name.